app/models/ep_postmaster/smtp_error.rb

```diff
@@ -0,0 +1,38 @@
+module EpPostmaster
+  class SMTPError
+
+    SMTP_ERROR_MESSAGE_ROLLUP = [
+      "The email address is invalid or there is an issue with the recipients mailbox",
+      "There is an issue with the recipients mailbox",
+      "There is an issue with the recipients mail server",
+      "There was a network error while delivering the email",
+      "There was an error sending the message",
+      "There is an issue with an attachment",
+      "There was an error sending the message"
+    ].freeze
+
+    # Mailgun unfortunately does not give us the enhanced status code neatly.
+    # It looks like it stuffs what it gets back from the desitnation server
+    # into the error field of the event record it sends via webhook.
+    # This will attempt to glean the enhanced code from that message.
+    # Sometimes the status code will be the enhanced code.
+    def self.normalize(status_code, error, recipient)
+      # 6xx is a Mailgun code, error contains the description
+      return error if status_code.to_s.match(/^6/)
+
+      unknown_error_message = "An error occured"
+
+      enhanced_code_pattern = /(\d\.\d\.\d)/
```

Comment:
⬇️ If you want this to be a little more readable down the line you could name your captures:
```
enhanced_code_pattern = /((?<class>\d)\.(?<subject>\d)\.(?<detail>\d))/
```
Then after a match:
`smtp_enhanced_status_code[:subject]` instead of having to `to_s.split(".")[arbitrary_position]`